homeassistant/components/owlet/sensor.py

```diff
@@ -0,0 +1,151 @@
+"""Support for Owlet sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    UnitOfTemperature,
+    UnitOfTime,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import DOMAIN, SLEEP_STATES
+from .coordinator import OwletCoordinator
+from .entity import OwletBaseEntity
+
+
+@dataclass
+class OwletSensorEntityDescription(SensorEntityDescription):
+    """Represent the owlet sensor entity description."""
+
+
+SENSORS: tuple[OwletSensorEntityDescription, ...] = (
+    OwletSensorEntityDescription(
+        key="battery_percentage",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    OwletSensorEntityDescription(
+        key="oxygen_saturation",
+        translation_key="o2saturation",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:leaf",
+    ),
+    OwletSensorEntityDescription(
+        key="oxygen_10_av",
+        translation_key="o2saturation10a",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:leaf",
+    ),
+    OwletSensorEntityDescription(
+        key="heart_rate",
+        translation_key="heartrate",
+        native_unit_of_measurement="bpm",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:heart-pulse",
+    ),
+    OwletSensorEntityDescription(
+        key="battery_minutes",
+        translation_key="batterymin",
+        native_unit_of_measurement=UnitOfTime.MINUTES,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.MEASUREMENT,
```

Comment:
```suggestion
```

This is a prediction and not a measurement, as the remaining minutes will be calculated